File: 0090-subsets-ii/0090-subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        
        nums.sort()
        res = set()
        res.add(tuple())
        
        for n in nums:
            cur_subset = set()
            
            for sub in res:
                cur_subset.add(tuple(list(sub) + [n]))
            
            res.update(cur_subset)
        
        return [list(a) for a in res]

File: 0090-subsets-ii/0090-subsets-ii.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsetsWithDup = function(nums) {
    nums.sort((a, b) => a - b)
    const res = []
    helper(0, [])
    return res
    
    function helper(idx, cur) {
        res.push(cur)
        for (let i=idx; i < nums.length; i++) {
            if (i > idx && nums[i-1] === nums[i]) continue
            helper(i+1, [...cur, nums[i]])
        }
    }
};